src/modules/70-pipeline/components/PipelineStudio/PipelineStudio.tsx

```diff
@@ -6,133 +6,126 @@
-              <String stringID="clickHere" />
-            </Button>
-            <Text>
-              <String stringID="errorHelp" />
-            </Text>
-          </Layout.Horizontal>
-          {__DEV__ && (
-            <React.Fragment>
-              <Text font="small">Error Message</Text>
-              <Container>
-                <details>
-                  <summary>Stacktrace</summary>
-                  <pre>{error.stack}</pre>
-                </details>
-              </Container>
-            </React.Fragment>
-          )}
-        </Layout.Vertical>
-      )
-    }
-    const {
-      className = '',
-      routePipelineStudio,
-      routePipelineDetail,
-      routePipelineList,
-      routePipelineProject,
-      getOtherModal
-    } = this.props
-    return (
-      <PipelineSchemaContextProvider>
-        <GitSyncStoreProvider>
-          <div className={cx(css.container, className)}>
-            <PipelineCanvas
-              // diagram={diagram}
-              toPipelineStudio={routePipelineStudio}
-              toPipelineDetail={routePipelineDetail}
-              toPipelineList={routePipelineList}
-              toPipelineProject={routePipelineProject}
-              getOtherModal={getOtherModal}
-            />
-          </div>
-        </GitSyncStoreProvider>
-      </PipelineSchemaContextProvider>
+  const { modal } = useQueryParams<{ modal?: ModuleLicenseType }>()
+
+  const getOtherModal = modal ? getTrialPipelineCreateForm : undefined
+  const handleRunPipeline = (): void => {
+    history.push(
+      routes.toPipelineStudio({
+        accountId,
+        orgIdentifier,
+        projectIdentifier,
+        pipelineIdentifier,
+        module,
+        branch,
+        repoIdentifier,
+        repoName,
+        connectorRef,
+        storeType,
+        runPipeline: true
+      })
     )
   }
+  const { licenseInformation } = useLicenseStore()
+  const isCFEnabled = useFeatureFlag(FeatureFlag.CFNG_ENABLED)
+  const isCIEnabled = useFeatureFlag(FeatureFlag.CING_ENABLED)
+  const isCDEnabled = useFeatureFlag(FeatureFlag.CDNG_ENABLED)
+  const isSTOEnabled = useFeatureFlag(FeatureFlag.SECURITY_STAGE)
+  const isPipelineChainingEnabled = useFeatureFlag(FeatureFlag.PIPELINE_CHAINING)
+  const { getString } = useStrings()
+
+  return (
+    <PipelineProvider
+      stagesMap={stagesCollection.getAllStagesAttributes(getString)}
+      queryParams={{
+        accountIdentifier: accountId,
+        orgIdentifier,
+        projectIdentifier,
+        repoIdentifier,
+        branch,
+        repoName,
+        connectorRef,
+        storeType
+      }}
+      pipelineIdentifier={pipelineIdentifier}
+      renderPipelineStage={args =>
+        getPipelineStages({
+          args,
+          getString,
+          module,
+          isCIEnabled: licenseInformation['CI'] && isCIEnabled,
+          isCDEnabled: licenseInformation['CD'] && isCDEnabled,
+          isCFEnabled: licenseInformation['CF'] && isCFEnabled,
```

Comment:
Any reason to not use the usual `.` object syntax here?
```suggestion
          isCIEnabled: licenseInformation.CI && isCIEnabled,
          isCDEnabled: licenseInformation.CD && isCDEnabled,
          isCFEnabled: licenseInformation.CF && isCFEnabled,
```